testing/test_warnings.py

```diff
@@ -169,7 +169,7 @@ def test_my_warning(self):
                     assert True
     """
     )
-    result = pytester.runpytest()
+    result = pytester.runpytest("-Werror")
```

Comment:
Hmm seems this one somewhat negates the purpose of the test, which IIUC is meant to pass no matter what. We should check this one... (Will try to do it later)